doc/languages-frameworks/go.section.md

```diff
@@ -0,0 +1,156 @@
+# Go
+
+## Go modules
+
+The function `buildGoModule` builds Go programs managed with Go modules.
+It builds a [Go Modules](https://github.com/golang/go/wiki/Modules) through a two phase build:
+- An intermediate fetcher derivation. This derivation will be used to fetch all of the dependencies of the Go module.
+- A final derivation will use the output of the intermediate derivation to build the binaries and produce the final output.
+
+### Example for `buildGoModule`
+
+In the following is an example expression using `buildGoModule`, the following
+arguments are of special significance to the function:
+
+- `vendorSha256`: is the hash of the output of the intermediate fetcher
+   derivation. `vendorSha256` can also take `null` as an input. When `null` is
+   used as a value, rather than fetching the dependencies and vendoring them, we
+   use the vendoring included within the source repo. If you'd like to not have
+   to update this field on dependency changes, run `go mod vendor` in your
+   source repo and set `vendorSha256 = null;`
```

Comment:
It would be nice not to hard-wrap the lines. All modern editors (even stuff like nano) have soft-wrapping but tools like `git diff` cannot handle hard-diffing.

Reply:
Is there some tool for auto-formatting markdown?

Reply:
IIRC prettier allows formatting markdown, but it also does hard-wrapping by default :nauseated_face:

Reply:
Try `nix shell -f channel:nixos-unstable nodePackages.prettier -c prettier --prose-wrap never --write doc/languages-frameworks/go.section.md`

Reply:
That formatter seemed to work for me for the Ruby docs in #105200

Reply:
done.